lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -457,6 +498,137 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
             # although most likely jobs aren't operating on the same object
             self._check_ws_objects(source_objects=job.get(_SOURCE_WS_OBJECTS))
 
+    def retry_multiple(
+        self, job_ids, as_admin=False
+    ) -> List[Dict[str, Union[str, Any]]]:
+        """
+        #TODO Add new job requirements/cgroups as an optional param
+        :param job_ids: The list of jobs to retry
+        :param as_admin: Run with admin permission
+        :return: The child job ids that have been retried or errors
+        """
+        retried_jobs = []
+        for job_id in job_ids:
```

Comment:
Hmm, I think there's a case to be made to do the error checking for all the jobs before actually making any DB changes or submitting jobs to condor.

Reply:
done

Reply:
So each of these retry job requests call the retry method.  The retry method does `_validate_retry_presubmit` and `_get_run_job_params_from_existing_job` where it validates the job permissions, job state is retryable, and not a batch job parent.  
Then it calls `run_job` which actually does state changes in the db. If it succeeds great, if not, it fails. 

If retry_job does all of the error checking for all of the jobs before submit/db writes, and then retry_jobs just calls retry_job therefore retry_jobs eventually does all of the error checking for jobs before submit

Reply:
Ok, what I was thinking was that you'd call `_validate...` on every job before calling `run_job` even once so that the retry call is all or nothing. I'm not sure it makes sense to return partial results in this case - any error caught in `_validate...` is going to be a client programming error and so indicate a bug AFAICT. To my mind it's better to fail the batch in that sort of case, again unless there's something I'm missing here.

Reply:
Ok done, validation from client errors now prevents the entire batch from submitting. Errors further down the line still result in partial submissions suceeding

Reply:
:+1: